opennms-services/src/main/java/org/opennms/netmgt/scriptd/Executor.java

```diff
@@ -202,87 +209,102 @@ public void run() {
                 }
             }
 
-            Script[] attachedScripts = m_event.getScript();
+            if (m_config.getTransactional()) {
+                m_sessionUtils.withReadOnlyTransaction(() -> {
```

Comment:
@agalue any opinion on whether we make this read-only or not? Read-only will have less of an impact on the system but this mean that you'll continue to have to manage your own transaction if you want to create or update any database entities.

Reply:
I know for a fact that there are customers that need to modify nodes from within Scriptd. They are currently doing that within a transaction using Groovy, meaning that the `read-only` nature should not affect them if applied.

For this reason, I think it makes sense to enforce `read-only`, as most of the times that object should just provide more context while processing the event.

Reply:
Thanks. We can keep it to read-only then as this will make sure that the LazyInitExceptions no longer occur. Changing objects will require the script developer to manage their own lookups and transactions though (which is the current case).